test_helpers.py

```diff
@@ -14,106 +14,119 @@
 
 # TODO(oschaaf):
 http = urllib3.PoolManager()
-pretty_print = pprint.PrettyPrinter(indent=4)
-
-def get_url(url, requestHeaders = {}, log=True, userAgent = None, timeout=None):
-  assert url.startswith("http")
-
-  copyRequestHeaders = copy(requestHeaders)
-  if timeout is None:
-      # TODO(oschaaf): sensible defaults
-      timeout = urllib3.Timeout(total=10.0)
-  if "User-Agent" in requestHeaders:
-    print requestHeaders["User-Agent"]
-    assert userAgent is None
-  else:
-    if userAgent is None:
-      userAgent = test_fixtures.DEFAULT_USER_AGENT
-    copyRequestHeaders["User-Agent"] = userAgent
-
-  resp = http.request('GET', url, headers=copyRequestHeaders, timeout=timeout, retries = False)
-  body = resp.data
-
-  if log:
-    print "get_url(): %s -> %s (%s)" % (url, resp.status, copyRequestHeaders)
-    pretty_print.pprint(resp.getheaders())
-    print "^^^^^^^^^^^^^^^^^^^^^"
-    print body
-  return resp, body
+pretty_print = pprint.PrettyPrinter(indent = 4)
+
+
+def get_url(url, requestHeaders = None, log = True, userAgent = None,
+            timeout=None):
+    assert url.startswith("http")
+    if requestHeaders is None:
+        requestHeaders = {}
+    copyRequestHeaders = copy(requestHeaders)
+    if timeout is None:
+        # TODO(oschaaf): sensible defaults
+        timeout = urllib3.Timeout(total=10.0)
+    if "User-Agent" in requestHeaders:
+        print requestHeaders["User-Agent"]
+        assert userAgent is None
+    else:
+        if userAgent is None:
+            userAgent = test_fixtures.DEFAULT_USER_AGENT
+        copyRequestHeaders["User-Agent"] = userAgent
+
+    resp = http.request(
+        'GET', url, headers = copyRequestHeaders, timeout = timeout,
+        retries=False)
```

Comment:
@jeffkaufman sounds good. currently this uses urllib3 because that is threadsafe. Perhaps we should add tests to verify the behaviour of the used http libs as well for bad responses? these types of comments are very helpful, thanks. I'm making large changes to this code set to get it acceptable per style guide.